lib/sinon/default-behaviors.js

```diff
@@ -286,6 +287,20 @@ const defaultBehaviors = {
 
     value: function value(fake, newVal) {
         const rootStub = fake.stub || fake;
+        const propertyDescriptor = getPropertyDescriptor(
+            rootStub.rootObj,
+            rootStub.propName,
+        );
+        const propertyIsGetter = propertyDescriptor.get !== undefined;
+        const propertyIsMethod =
+            !propertyIsGetter &&
+            typeof rootStub.rootObj[rootStub.propName] === "function";
+
+        if (propertyIsMethod) {
+            throw new Error(
+                `${rootStub.propName} is a function, not a getter or value. Use .returns() instead of .value()`,
```

Comment:
Are we supposed to allow setting `.value()` on a getter property? Seems erronous.